server/api.js

```diff
@@ -42,4 +42,25 @@ router.delete("/videos/:id", async (req, res) => {
 	}
 });
 
+router.put("/videos/:id/rating", async (req, res) => {
+	const videoId = req.params.id;
+	const { rating } = req.body;
+
+	try {
+		const result = await db.query(`SELECT * FROM videos WHERE id=${videoId}`);
+		let video = result.rows[0];
+		if (!video) {
+			return res.status(404).json({ error: "Video not found" });
+		}
+
+		await db.query(`UPDATE videos SET rating=${rating} WHERE id=${videoId}`);
+
+		res
+			.status(200)
+			.json({ success: true, message: "Video rating successfully" });
+	} catch (error) {
+		res.status(500).json({ success: false, error: "Internal server error" });
```

Comment:
It's good that you're using an appropriate error code, but I would like  to see you being  more specific here, yes an internal server error happened but why?